### PR TITLE
Automate figma documentation

### PR DESCRIPTION
**Pull request by lukasoppermann**
This is the very beginning of adding figma to the new docs.

[👀 Preview](https://primer-822a6e3645-26441320.drafts.github.io/components/action-bar/figma)

## Changes
- removes figmaUrl and replaced with figmaId
- store figma data in graphQL
- provide data to components through props

## Todo
- [x] currently data is loaded in the components from an npm module, this should be replaced so that it is available via gatsby @josepmartins @danielguillan can you do this?
- [x] currently `component.thumbnails.prop` is an object with dynamic props. We should replace this with an array of entries `['propertyName', 'propertyValue']` so that it is graphQL compatible @danielguillan @josepmartins 
- [x] some components, like `actionList` are not available. I think this may be, because it is not a component set CC: @danielguillan?
- [x] Pictures don't show up, this is either figma and they are gone, or an issue with access rights / cors, etc. @colebemis any idea?
- [x] individual properties should be added automatically, we need to have a list of all properties in gatsby to loop over the props and add the components.
- [x] somehow the anchor nav does not show up
- [x] fix inconsistent spacing below headlines
- [x] images don't show up
- [x] we should probably remove the last edited by section in the overview, right?
- [x] spacing in overview (to section below yellow note) should be adjusted to fit to react page
- [x] components should be cleaned up and named better.

### CODE REVIEW

**Comment by emilybrick** [comment]
Thank you for taking this on @lukasoppermann ! 👏 🙏 so excited to have all component docs in one place. 

Questions about task list, what do these two refer to? 
- spacing in overview (to section below yellow note) should be adjusted to fit to react page
- components should be cleaned up and named better.

**Comment by lukasoppermann** [comment]
> - spacing in overview (to section below yellow note) should be adjusted to fit to react page
This is partially improved. It is the spacing between the badge and the variants at the very top

> - components should be cleaned up and named better.
This is just how the figma react components are named and build

**Comment by colebemis** [comment]
@lukasoppermann The image issue looks like a permissions problem:

CleanShot 2023-02-28 at 18 33 35@2x

Where you and @josepmartins able to get the images working in another prototype?

**Comment by lukasoppermann** [comment]
> @lukasoppermann The image issue looks like a permissions problem:
> 
> CleanShot 2023-02-28 at 18 33 35@2x

This is actually normal for figma images. They don't allow browser get somehow. It used to work here: https://primer-af96a7b598-26441320.drafts.github.io/guides/figma/components/banner

But it is broken there to. I think @josepmartins and @danielguillan will have a look once josep is back from his vacation.

**Comment by josepmartins** [comment]
@lukasoppermann @danielguillan and I made the latest changes and its ready for review

File: content/components/checkbox-group.mdx [hunk 1]
```diff
@@ -2,7 +2,7 @@
 title: Checkbox group
 description: Checkbox group renders a set of checkboxes.
 reactId: checkbox_group
-figma: https://www.figma.com/file/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?node-id=15341%3A47574
+figmaId: checkbox
```
Comment (danielguillan) [inline content/components/checkbox-group.mdx:5]: Is it intentional to document `checkbox` in `checkbox_group` for Figma?

File: gatsby-node.js [hunk 2]
```diff
@@ -124,6 +125,53 @@ async function sourceOcticonData({actions, createNodeId, createContentDigest}) {
...
+  }
+
+  const newNode = {
+    ...nodeData,
+    id: createNodeId('figma-file'),
+    internal: {
+      type: 'FigmaFile',
+      contentDigest: createContentDigest(nodeData),
+    },
+  }
```
Comment (danielguillan) [inline gatsby-node.js:170]:
```suggestion
  const newNode = {
    fileUrl,
    id: createNodeId('figma-file'),
    internal: {
      type: 'FigmaFile',
      contentDigest: createContentDigest(nodeData),
    },
  }
```

File: src/components/FigmaPropertyPreview.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+export default function FigmaPropertyPreview({
+  thumbnails,
+  property,
+  setProperties = {},
+  column = undefined,
+  hideLabels = false,
+}) {
+  const previewComponents = getPreviewComponents(thumbnails, property, setProperties)
+  console.log(property, thumbnails, previewComponents)
```
Comment (danielguillan) [inline src/components/FigmaPropertyPreview.js:38]:
```suggestion

```

File: content/components/flash.mdx [hunk 4]
```diff
@@ -2,7 +2,7 @@
 title: Flash
 description: Flash alert informs users of successful or pending actions.
 reactId: flash
-figma: https://www.figma.com/file/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?node-id=142%3A597&t=PFYau6cmDlujeJkx-1
+figmaId: banner
```
[the remark this answers is not in the document]
Comment (josepmartins) [reply content/components/flash.mdx:5]: Yes, it is planned to worked on naming and fix other accessibility issues in the PRC `Flash`: https://github.com/github/primer/issues/1824

File: content/components/checkbox-group.mdx (hunk 1 above)
Comment (josepmartins) [reply content/components/checkbox-group.mdx:5]:
```suggestion
```

File: gatsby-node.js [hunk 5]
```diff
@@ -124,6 +125,53 @@ async function sourceOcticonData({actions, createNodeId, createContentDigest}) {
...
+  const {fileUrl, components} = json
+
+  /**
+   * Add figma components to the GraphQL store
+   */
+  for (const component of components) {
+    component.thumbnails = component.thumbnails.map(thumbnail => {
+      thumbnail.props = thumbnail.props.map(({name, value}) => [name, value])
+      return thumbnail
+    })
```
[the remark this answers is not in the document]
Comment (lukasoppermann) [reply gatsby-node.js:143]: I had some issue without doing this, but I am sure we can work around this somehow.

File: gatsby-node.js [hunk 6]
```diff
@@ -124,6 +125,49 @@ async function sourceOcticonData({actions, createNodeId, createContentDigest}) {
   }
 }
 
+async function sourceFigmaData({actions, createNodeId, createContentDigest}) {
+  // Save the Primer Figma data to the GraphQL store
+  const json = await fetch(
+    `https://raw.githubusercontent.com/primer/figma/main/packages/web/generated/components.json`,
```
Comment (colebemis) [inline gatsby-node.js:131]: How does this data get updated?

File: src/components/FigmaComponentPlayground.js
Comment (colebemis) [inline src/components/FigmaComponentPlayground.js:1]: Could you rename these files to kebab case (`figma-component-playground.js`) to match the other files in this directory?

**Review by colebemis — APPROVED** [review]
This looks amazing! 🚀

File: gatsby-node.js (hunk 6 above)
Comment (danielguillan) [reply gatsby-node.js:131]: We've set up [a workflow](https://github.com/primer/figma/blob/main/.github/workflows/docgen-web.yml) that updates the JSON file daily (can be triggered on demand too). Also, we're planning to hook it up to the Figma library release process that is being worked on, so it's also triggered automatically when a new release is published.

**Pull request merged**